src/item_box.py

```diff
@@ -49,7 +49,7 @@ def on_dnd_drop(self, drop_target, value, _x, _y):
             self.parent.datadict, drop_target.key, value)
         ui.reload_item_menu(self.parent)
         clipboard.add(self.parent)
-        graphs.reload(self.parent)
+        graphs.refresh(self.parent)
```

Comment:
I'm not sure if this line is the exact culprit, but when a file is loaded that has a header, it does not actually load the new label directly. This is because the graph needs to be re-rendered before text items are changed. So you need to e.g. resize the window first before that effect is taken into place.

So at least when a title/axis label is changed, it should probably reload the graph completely.